trusted_applet/internal/update/update.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Armored Witness Applet authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package update provides functionality for fetching updates, verifying
+// them, and installing them onto the armory device.
+package update
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/golang/glog"
+	"github.com/transparency-dev/armored-witness-applet/trusted_applet/internal/firmware"
+)
+
+// Local allows access to query the firmware installed on this device and
+// operations to install new versions of the firmware.
+type Local interface {
+	// GetInstalledVersions returns the semantic versions of the OS and Applet
+	// installed on this device. These will be the same versions that are
+	// currently running.
+	GetInstalledVersions() (os, applet semver.Version, err error)
+
+	// InstallOS updates the OS to that included in the firmware bundle.
+	InstallOS(firmware.Bundle) error
+
+	// InstallApplet updates the Applet to that included in the firmware bundle.
+	InstallApplet(firmware.Bundle) error
+
+	// Reboot instructs the device to reboot after new firmware is installed.
```

Comment:
I read this to mean that the function will hang. Do you mean that, or do you mean to say that there is no return value?

Reply:
Understood now, also thanks for the added comment.